pkg/recipes/terraform/config/providers/aws.go

```diff
@@ -139,10 +151,33 @@ func (p *awsProvider) generateProviderConfigMap(credentials *credentials.AWSCred
 		config[awsRegionParam] = region
 	}
 
-	if credentials != nil && credentials.AccessKeyCredential != nil &&
-		credentials.AccessKeyCredential.AccessKeyID != "" && credentials.AccessKeyCredential.SecretAccessKey != "" {
-		config[awsAccessKeyParam] = credentials.AccessKeyCredential.AccessKeyID
-		config[awsSecretKeyParam] = credentials.AccessKeyCredential.SecretAccessKey
+	if credentials != nil {
+		switch credentials.Kind {
+		case ucp_datamodel.AWSAccessKeyCredentialKind:
+			if credentials.AccessKeyCredential != nil &&
+				credentials.AccessKeyCredential.AccessKeyID != "" && credentials.AccessKeyCredential.SecretAccessKey != "" {
+				config[awsAccessKeyParam] = credentials.AccessKeyCredential.AccessKeyID
+				config[awsSecretKeyParam] = credentials.AccessKeyCredential.SecretAccessKey
+			}
+
+		case ucp_datamodel.AWSIRSACredentialKind:
+			// Radius requests will first be routed to STS endpoint,
+			// where it will be validated and then the request to the specific service (such as S3) will be made using
+			// the bearer token from the STS response.
+			// Based on the https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html,
+			// STS endpoint should be region based, and in the same region as
+			// Radius instance to minimize latency associated with STS call and thereby improve performance.
```

Comment:
This comment seems out of place because the code below isn't making any calls to AWS, was this intended?

Reply:
I have added "once we switch to region based STS endpoint, we should add the region to the config." to the comment. Hope that makes it clearer. issues/7747 is at a high level going to introduce a way to set region for both ucp and terraform, based on radius's location.